kratos/processes/simple_mortar_mapper_process.h

```diff
@@ -0,0 +1,474 @@
+                                
+                                // Integrating the mortar operators
+                                for ( unsigned int point_number = 0; point_number < integration_points_slave.size(); point_number++ )
+                                {
+                                    const PointType local_point_decomp = integration_points_slave[point_number].Coordinates();
+                                    PointType local_point_parent;
+                                    PointType gp_global;
+                                    decomp_geom.GlobalCoordinates(gp_global, local_point_decomp);
+                                    slave_geometry.PointLocalCoordinates(local_point_parent, gp_global);
+           
+                                    /// SLAVE CONDITION ///
+                                    slave_geometry.ShapeFunctionsValues( rVariables.NSlave, local_point_parent.Coordinates() );
+                                    rVariables.PhiLagrangeMultipliers = rVariables.NSlave;
+                                    rVariables.DetjSlave = slave_geometry.DeterminantOfJacobian( local_point_parent );
+                                    
+                                    /// MASTER CONDITION ///
+                                    PointType projected_gp_global;
+                                    const array_1d<double,3> gp_normal = MortarUtilities::GaussPointNormal(rVariables.NSlave, slave_geometry);
+                                    
+                                    GeometryType::CoordinatesArrayType slave_gp_global;
+                                    slave_geometry.GlobalCoordinates( slave_gp_global, local_point_parent );
+                                    MortarUtilities::FastProjectDirection( master_geometry, gp_global, projected_gp_global, master_normal, -gp_normal ); // The opposite direction
+                                    
+                                    GeometryType::CoordinatesArrayType projected_gp_local;
+                                    
+                                    master_geometry.PointLocalCoordinates(projected_gp_local, projected_gp_global.Coordinates( ) ) ;
+                                    
+                                    // SHAPE FUNCTIONS 
+                                    master_geometry.ShapeFunctionsValues( rVariables.NMaster, projected_gp_local );    
+                                    
+                                    const double integration_weight = integration_points_slave[point_number].Weight();
+                                    
+                                    rThisMortarConditionMatrices.CalculateMortarOperators(rVariables, integration_weight);   
+                                }
+                            }
+                        }
+                        
+                        // We compute the norm
+                        const double norm_D = norm_frobenius(rThisMortarConditionMatrices.DOperator);
```

Comment:
What are u doing here? The mortar matrix is a global, sparse matrix , you can not invert it locally.